server.py

```diff
@@ -45,85 +57,145 @@ def __init__(self, map_name):
         await ws.prepare(request)
         # WebSocket is added to a list
         ws_list.append(ws)
-
         try:
             yield ws
         finally:
             # after disconnection client is removed from list
             ws_list.remove(ws)
 
-    async def receiver(self, request):
+    async def talk_to_receiver(self, request):
+        """
+        Communicate with websockets connected through `/receiver/` route.
+
+        Send them game state.
+        Maintain connection to the client until they disconnect.
+        """
         async with self.ws_handler(request, self.ws_receivers) as ws:
             # This message is sent only this (just connected) client
             await ws.send_json(self.state.as_dict(map_name))
             # For cycle keeps the connection with client alive
-            async for msg in ws:
+            async for message in ws:
                 pass
             return ws
 
-    async def interface(self, request):
+    async def talk_to_interface(self, request):
+        """
+        Communicate with websockets connected through `/interface/` route.
+
+        Send them their robot name, game state and cards to choose.
+        React to the messages from interface: update game state accordingly.
+        Maintain connection to the client until they disconnect.
+        """
         async with self.ws_handler(request, self.ws_interfaces) as ws:
-            # This message is sent only this (just connected) client
-            # Client_interface is added to dictionary (robot.name: ws)
-            name = self.available_robots[0].name
-            self.assigned_robots[name] = ws
-            print(self.assigned_robots)
+            ##################################################################
+            # SERVER PART ONE: communicate only with this client             #
+            ##################################################################
+
+            # Get first data for connected client: robot and cards
+            robot = self.assign_robot_to_client(ws)
+            cards = self.state.get_dealt_cards(robot)
+
+            # Prepare message to send: robot name, game state and cards
+            welcome_message = {"robot_name": robot.name,
+                               **self.state.as_dict(map_name),
```

Comment:
co znamenají ty dvě hvězdičky?

Reply:
je to spojovani slovniku, jak jsme se bavili na srazu 2. 7, zápis je připraven zde: https://github.com/PyLadiesCZ/roboprojekt-blog/pull/44/files

Reply:
Díky, jsem si říkala že jsem to někde už viděla  😁